CORDA-3193: Update docs to mention branching strategy

lockathan opened the pull request:
Removed mention of master branch except to describe that the master branch is no longer used.
Added explanation of branching strategy to contributing guide.
Renamed building-against-master to building-against-non-release

lockathan committed:
CORDA-3193: Update docs to mention branching strategy

Removed mention of master branch except to describe that the master branch is no longer used.
Added explanation of branching strategy to contributing guide.
Renamed building-against-master to building-against-non-release

gendal commented:
for an external contributor who may not know the answers to these questions, what do we suggest?  open against the default branch? contact us first?

blsemo commented on docs/source/contributing.rst:
```diff
@@ -47,11 +47,26 @@ Making the required changes
+`release/os/{major version}.{minor version}`
+
+Note that `release` is always part of the name of the branch, even for unreleased versions of Corda. The default 
+github branch will always be the current development branch of Corda. Development work should target the oldest version 
```
As @gendal said - mention here that this isn't just the default GitHub branch, it should also be the default to work against (unless fixing a specific issue in a specific version).

blsemo commented on docs/source/contributing.rst:
```diff
@@ -117,7 +132,7 @@ Merging the changes back into Corda
 You should merge the changes back into Corda as follows:
 
-1. Create a pull request from your fork to the ``master`` branch of the Corda repo
+1. Create a pull request from your fork to the current development branch of the Corda repo
```
What is the `current development branch`? Shouldn't that be the release branch for the version you developed against?

lockathan committed:
CORDA-3193: Update docs to mention branching strategy

Updated contributing document to mention that work should go into the default branch by default.

edward-prosser commented on docs/source/contributing.rst:
```diff
@@ -105,7 +121,7 @@ You should document any changes to Corda's public API as follows:
    synonyms (e.g. ``%d`` and ``%date``), we prefer the longer form for legibility reasons. You can configure your IDE
    to highlight these in bright yellow
-2. Update the relevant `.rst file(s) <https://github.com/corda/corda/tree/master/docs/source>`_
+2. Update the relevant `.rst file(s)
```
Remove the backtick from before .rst

lockathan committed:
CORDA-3193: Update docs to mention branching strategy

Removed unnecessary backtick.

lockathan committed:
CORDA-3193: Updated contributing section

Corrected term from "current development branch" to "equivalent branch"

blsemo reviewed (approved):
LGTM
